crypto/rand/rand_lib.c

```diff
@@ -354,6 +360,20 @@ void rand_cleanup_int(void)
     rand_nonce_lock = NULL;
 }
 
+/*
+ * RAND_close_seed_files() ensures that any seed file decriptors are
+ * closed after use.
+ */
+void RAND_set_close_seed_files(int close)
+{
+    rand_pool_set_close_seed_files(close);
+}
+
+int RAND_get_close_seed_files(void)
```

Comment:
As a person who likes self describing function names in imperative style (`CLASSNAME_do_this()`, `CLASSNAME_enable_that()`), I would prefer to have a more descriptive function name than "this is the setter of some internal variable called `close_seed_files`. Even more since

- There is no need for a setter/getter pair. This feature is under control of the application and nobody else should be changing it anyway. So there is no need for the application to query the current value.
- The term "seed files" is used nowhere else and is unnecessarily general; everywhere else we are talking about  random device(s). Using such a general term just hides information to the programmer, which has no clue what precisely this function is doing unless he looks up the documentation or the source code.

My suggestion: Just drop the getter and rename the setter to something like

```
void RAND_keep_devrandom_open(int keep);
void RAND_keep_random_devices_open(int keep);
```

or the other way around

```
void RAND_close_devrandom_after_use(int close);
void RAND_close_random_devices_after_use(int close);
```

Reply:
I often admin I'm rubbish with API names :)  I'll change it.

Reply:
Please don't use `devrandom` in the names.  Sure, this is *currently* very Unix centric, but should this become implemented on other platforms as well, `devrandom` is as foreign a name as it gets.

Reply:
This function exists solely to disable a workaround for a very unix-ish problem, namely that /dev/random might not be available after chroot(). So it is very unlikely that it will ever be used in a more general context.

Reply:
> Please don't use `devrandom` in the names.

Does this include `random_devices`, too, or only `devrandom`?

Reply:
`devrandom` specifically. `random_devices` is fine in my mind.

Reply:
Renamed to `RAND_keep_random_devices_open` and removed the getter.